src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -353,10 +385,76 @@ public function checkAnnotations(\ReflectionClass $refl, string $class): array
                 }
             }
 
+            if (!($hasDocReturnType = false !== strpos($method->getDocComment(), '@return') && preg_match('/\n\s+\* @return +(\S+)/', $doc, $matches)) && $parentsAndInterfacesThatHaveMethodsThatReturnsADocType && !$method->hasReturnType()) {
+                foreach ($parentsAndInterfacesThatHaveMethodsThatReturnsADocType as $name) {
+                    if (!isset(self::$returnTypes[$name][$method->name])) {
+                        continue;
+                    }
+
+                    list($normalizedDocReturnType, $hintedReturnType) = self::$returnTypes[$name][$method->name];
+                    $deprecations[] = sprintf('The "%s::%s()" parent method returns %s. You could safely add the ": %s" return type to your method to be compatible with the next major version of its parent.', $class, $method->name, $normalizedDocReturnType, $hintedReturnType);
```

Comment:
what about this?
`'The parent method of "%s::%s()" uses an annotation that declares it returns %s and might be turned into a native return type as of its next major version. You need to add the "%s" return type to your method to keep being compatible.'`